### PR TITLE
Az655 osync warning

### PR DESCRIPTION
**Pull request by massung**
Requires branch az655-osync-warning from riak_core as well. On linux, checks for bitcask config {sync_strategy,o_sync} and warns (if on Linux) that it will fail.

### CODE REVIEW

File: src/riak_kv_bitcask_backend.erl [hunk 1]
```diff
@@ -89,6 +89,7 @@ start(Partition, Config) ->
                     BitcaskOpts = [{read_write, true} | Config],
                     case bitcask:open(filename:join(DataRoot, DataFile), BitcaskOpts) of
                         Ref when is_reference(Ref) ->
+			    check_fcntl(),
```
Comment (rzezeski) [inline src/riak_kv_bitcask_backend.erl]: Substitute hard tabs for spaces.

File: src/riak_kv_bitcask_backend.erl [hunk 2]
```diff
@@ -309,6 +310,25 @@ key_counts(RootDir) ->
...
+%% On linux there is a kernel bug that won't allow fcntl to add O_SYNC
+%% to an already open file descriptor.
+check_fcntl() ->
+    Logged=application:get_env(riak_kv,o_sync_warning_logged),
+    Strategy=application:get_env(bitcask,sync_strategy),
+    case {Logged,Strategy} of
+	{undefined,{ok,o_sync}} ->
+	    case riak_core_util:is_arch(linux) of
+		true ->
+		    lager:warning("{sync_strategy,o_sync} fails on Linux"),
```
Comment (jonmeredith) [inline src/riak_kv_bitcask_backend.erl]: I would say 'not implemented' rather than fails.

**Comment by rzezeski** [comment]
So why not put this check in `riak_kv_app:start/2` instead and avoid the "check if I already logged this" code?

**Comment by jonmeredith** [comment]
Because it's a pluggable backend.  It's in the right place.

**Comment by rzezeski** [comment]
+1 to merge

**Pull request merged**